File: Groupe1/TP2/src/bd_école.c
#include <stdio.h>
#include <string.h>

// Définition de la structure Etudiant
struct Etudiant {
    char nom[100];
    char prenom[100];
    char adresse[200];
    float notes[2];
};


// Initialisation des données des étudiants avec des valeurs par défaut
void initialiserEtudiant(struct Etudiant *etudiant) {
    strcpy(etudiant->prenom, "Par défaut");  //Copie de la valeur "Par défaut" dans les éléments de la structure
    strcpy(etudiant->nom, "Par défaut");
    strcpy(etudiant->adresse, "Par défaut");
    //strcpy(etudiant->notes[0], 0.0);

    // Initialisation des notes
    for (int i = 0; i < 2; i++) {
        etudiant->notes[i] = 0.0;  //Initialisation des notes à 0
    }
}

// Affichage des informations d'un étudiant
void afficher_etudiant(struct Etudiant etudiant) {
    printf("Nom : %s\n", etudiant.nom); //Affiche le Nom
    printf("Prénom : %s\n", etudiant.prenom); //Affiche le prénom
    printf("Adresse : %s\n", etudiant.adresse); //Affiche l'adresse
    for (int i = 0; i < 2; i++) {
        printf("Note %d : %.2f\n", i + 1, etudiant.notes[i]); //Affiche les notes
    }
    printf("\n"); //retour à la ligne
}

int main() {
    struct Etudiant etudiants[5]; //Tableau pour stocker les 5 étudiant
    
    for (int i = 0; i < 5; i++) {
        initialiserEtudiant(&etudiants[i]); // Initialisation de chaque étudiant avec des valeurs par défaut
    }

    // Saisie par l'utilisateur des informations pour chaque étudiant
    for (int i = 0; i < 5; i++) {
        printf("Étudiant.e %d :\n", i + 1); //Affiche l'étudiant N°
        printf("Nom : "); // Affcihe ce que l'utilisateur doit rentrer
        scanf("%s", etudiants[i].nom); // Récupère la chaine de caracht"re rentré par l'utilisateur
        printf("Prénom : ");
        scanf("%s", etudiants[i].prenom);
        printf("Adresse : ");
        scanf("%s", etudiants[i].adresse);
        
        for (int j = 0; j < 2; j++) {
            printf("Note %d : ", j + 1); //Demande de rentrer la note 1 puis 2
            scanf("%f", &etudiants[i].notes[j]);
        }
        printf("\n"); //retoure à la ligne
    }
    
    // Affichage des informations des étudiants
    printf("\nAffichage des informations des étudiants \n");
    for (int i = 0; i < 5; i++) {
        printf("Étudiant.e %d :\n", i + 1); 
        afficher_etudiant(etudiants[i]); //Appelle de la fonction pour afficher toutes les informations de la liste d'étudiants
    }
    
    return 0; //return 0 
}

File: Groupe1/TP2/src/bd_école.py
# Définition de la classe Étudiant
class Etudiant:
    def __init__(self, nom, prenom, adresse, notes):
        self.nom = nom  #Créé le nom,prénom,adresse,notes
        self.prenom = prenom
        self.adresse = adresse
        self.notes = notes
    
    def afficher_etudiant(self): # Définition de la fonction pour afficher les étudiants
        print(f"Nom : {self.nom}")#Print le nom, prénom, adresse, notes 
        print(f"Prénom : {self.prenom}")
        print(f"Adresse : {self.adresse}")
        for i, note in enumerate(self.notes, 1):
            print(f"Note {i} : {note:.2f}")
        print()


etudiants = []  # Création du dictionnaire "étudiants"
    
# Saisie des informations pour chaque étudiant
for i in range(5):  #Boucle for pour les 5 étudiants
    print(f"Étudiant.e {i + 1} :") #Affiche du N° de l'étudiants
    nom = input("Nom : ")  #Demande à l'utilisateurs de rentrer le nom
    prenom = input("Prénom : ") 
    adresse = input("Adresse : ")
    notes = []
    for j in range(2):  # Pour les deux notes de l'étudiant
        note = float(input(f"Note {j + 1} : "))
        notes.append(note)
    etudiants.append(Etudiant(nom, prenom, adresse, notes)) #Ajout des information dans le dictionaire
    
    
# Affichage des informations des étudiants
for i, etudiant in enumerate(etudiants, 1): #Boucle pour afficher chaque étudiant
    print(f"Étudiant.e {i} :") #Affichage du n° d'étudiant
    etudiant.afficher_etudiant() #Appel la fonction défini dans la class